### DISCUSSION TITLE
getServerSideProps still called even I pass shallow true prop in next Link?

### DISCUSSION THREAD

**Discussion by vkotu** [vercel/next.js#45547] [Help]
I am passing shallow true prop true to the next Link, that doesn't prevent the getServerSideProps call during client-side to and fro navigation between pages. How to fix this problem? 
<img width="998" alt="image" src="https://user-images.githubusercontent.com/8981254/216522676-5a13f304-f2f6-4a89-8991-a5de1c2150b1.png">

**Comment by icyJoseph** [comment]
According to the docs:

> [shallow](https://nextjs.org/docs/routing/shallow-routing): Update the path of the current page without rerunning [getStaticProps](https://nextjs.org/docs/basic-features/data-fetching/get-static-props), [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props) or [getInitialProps](https://nextjs.org/docs/api-reference/data-fetching/get-initial-props). Defaults to false

This means that this applies only for navigation within the same page. This is useful for dynamic paths that upon navigation, change the URL, but are still using the same Next.js page. 

Working as expected.

If you want to skip GSSP data fetch on SPA transitions, but want to use it for "direct" navigation, there's a whole article on the subject. 

This is not trivial to do, and relies on Next.js internals, which I guess should be kind of stable, but since it is not part of the public API, it might break in future versions.

- the article: https://www.gregroz.me/article/nextjs-getServerSideProps-interception,
- and source code: https://github.com/Gelio/next-intercept-ssr-navigation/blob/main/src/intercept-getServerSideProps.ts

**Reply by vkotu** [reply]
Thanks for taking the time and share information. I will go through the article.

**Reply by icyJoseph** [reply]
Here's more from the docs: https://nextjs.org/docs/routing/shallow-routing
